Added deploy by name (#2331)
foundry-rs/starknet-foundry: docs/src/starknet/declare-deploy.md

> # Joining the Declaration and Deployment Together

Reviewer: Maybe this would be clearer to the users? Or something similar

```suggestion
# Deploying an Undeclared Contract
```

`sncast` allows declaring and deploying contracts through dedicated commands.

> The `declare-deploy` command simplifies this pipeline by performing these two actions one after the other. It is used to declare a contract and deploy it immediately.

Reviewer: ```suggestion
The `declare-deploy` command simplifies this pipeline by performing these two actions at the same time. It is used to declare a contract and deploy it immediately.
```

Reviewer: I know what you wrote is technically correct, but I think my suggestion is a better description for the user

For detailed description, see [declare-deploy command reference](../appendix/sncast/declare-deploy.md).

## Examples

Command operates under all the circumstances [`declare`](./declare.md) and [`deploy`](./deploy.md) would do when invoked separately.

### General Example

> Make sure you have a `Scarb.toml` in your project directory. Suppose we would like to declare and instantly deploy an example contract named `SimpleBalance`, defined in the default project.

Reviewer: I don't think we have a contract like that in template. Maybe let's use the name of the contract that's defined there?

Reviewer 2: Changed to `HelloSncast`

# Joining the Declaration and Deployment Together

> `sncast` allows declaring and deploying contracts through dedicated commands. However, sometimes contracts are meant to be only instantiated once, thus the standard way of declaration and deployment is unnecessarily time-consuming.

Reviewer: I'd just say it's an simpler solution when you want to immediately deploy a declared contract.

The `declare-deploy` command allows performing both actions at once.

For detailed description, see [declare-deploy command reference](../appendix/sncast/declare-deploy.md).

## Examples

> Command operates under all the circumstances `declare` and `deploy` would do when invoked separately.

Reviewer: Should this link to declare and deploy sections?

### General Example

Make sure you have a `Scarb.toml` in your project directory. Suppose we would like to declare and instantly deploy an example contract named `SimpleBalance`, defined in the default project.

Running:

```shell
$ sncast --account myuser \
    --url http://127.0.0.1:5050/rpc \ 
    declare-deploy \
    --fee-token strk \
    --contract-name SimpleBalance
```

results in declaration and deployment.

> > 📝 **Note**

Reviewer: Same here, I'd make it a warning
